open-api/rest-catalog-open-api.yaml

```diff
@@ -3103,6 +3103,79 @@ components:
         uuid:
           type: string
 
+    ADLSCredentials:
+      type: object
+      allOf:
+        - $ref: '#/components/schemas/Credentials'
+      required:
+        - type
+        - sas-token
```

Comment:
Is SAS the only supported credentials type of ADLS? Would it be worth adding sub-types (future-proofing)?

Reply:
E.g. `Account Key`? I'm not saying we should support it right away, but alternatives to SAS apparently exist.

Reply:
I believe vended credentials should have a set expiration on them and that's only possible for a SAS token, hence why there's no accountKey / accountName here as it's risky to leak such credentials and not have expiration on them. Let's move this discussion to the design doc (once I share it on the ML)

Reply:
I would agree with this and also state that we should never pass non-expiring credentials through this interface as it would be very dangerous from a security posture to expose unbounded credentials.  This means we should also require session token for AWS.

Reply:
I've added some wording that says all credentials need an expiration